### ISSUE TITLE
SC2046 when checking return code inline with command substitution

### ISSUE THREAD

**Issue by Speeddymon** [koalaman/shellcheck#792]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2046
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```
#!/bin/bash
if [ $(foo)$? -eq 1 ]
then
...
fi
```

#### Here's what shellcheck currently says:
```
Line 6:
if [ $(foo)$? -eq 1 ]
     ^-- SC2046: Quote this to prevent word splitting.
```

#### Here's what I wanted or expected to see:
Shellcheck should detect that $? is in use at the end of the command substitution and realize I am checking the return code of the command rather than the output.

In this particular case, I have realized I don't need to explicitly check for 1 return code. I can instead check for non-zero return code. So I have updated my code to use the below.

```
if ! foo
then
...
fi
```

Point is perhaps update shellcheck to give SC2181 in the above case due to $? at the end of the command substitution.

**Comment by koalaman** [comment]
Actually, you *are* checking the output. Just with the exit code concatenated to the end. 

`foo` outputs "hello world", you'll be doing `[ hello world1 -eq 1 ]` which will fail with an error. If it outputs "4" and returns 1, you'll be doing `[ 41 -eq 1 ]`, which will just give the wrong result.

It only happens to work if `foo` will never write anything to stdout, which is hard for `shellcheck` to determine.

The canonical way to do this without checking the output of `foo` would be:

```
foo
if [ $1 -eq 1 ]
```

**Comment by Speeddymon** [comment]
Thanks. I'll close this then.
